src/nvim/highlight.c

```diff
@@ -475,13 +476,43 @@ int hl_get_underline(void)
       .rgb_bg_color = -1,
       .rgb_sp_color = -1,
       .hl_blend = -1,
+      .url = STRING_INIT,
     },
     .kind = kHlUI,
     .id1 = 0,
     .id2 = 0,
   });
 }
 
+/// Augment an existing attribute with the beginning or end of a URL hyperlink.
+///
+/// @param attr Existing attribute to combine with
+/// @param url The URL to associate with the highlight attribute. An empty string indicates the
+///            end of a URL
```

Comment:
This might be how it works in the TUI but in the highlight layer it is better to just say an empty string indicates the _lack_ of an URL.

Reply:
This is out of date now and reflects a previous implementation, thanks for catching it. Will update the comment.

Reply:
b81970763e0734c6b5182e7b3608ad0a699e376e